src/test/groovy/net/rubygrapefruit/platform/file/BasicFileEventFunctionsTest.groovy

```diff
@@ -801,4 +801,34 @@ class BasicFileEventFunctionsTest extends AbstractFileEventFunctionsTest {
         def ex = thrown AbstractFileEventFunctions.FileWatcherTimeoutException
         ex.message == "Starting the watcher timed out"
     }
+
+    def "can detect events in directory removed then re-added"() {
+        given:
+        def watchedDir = new File(rootDir, "watched")
+        assert watchedDir.mkdirs()
+        def createdFile = new File(watchedDir, "created.txt")
+        startWatcher(watchedDir)
+
+        def directoryRemoved = watchedDir.delete()
+        def directoryRecreated = watchedDir.mkdirs()
+        if (!Platform.current().windows) {
```

Comment:
Can you add a comment why the directory can't be deleted on Windows?

Reply:
I'm not sure about the reason, but I agree a comment acknowledging that this is a known issue on Windows is in order.

Reply:
Followed up in https://github.com/gradle/native-platform/pull/204.